Extending ruleset with default off skips rule

**Describe the bug**
I had a ruleset that worked exactly how I wanted it to in 5.7.1. In upgrading to the latest 6.3.0, I noticed at least one of my rules isn't evaluating properly when running the cli. I am extending a ruleset that I have default to off. I extend a particular rule to modify it. In 5.7.1, the rule fails appropriately. In 6.3.0 it does not. Possibly a bug, possibly some sort of change in how I need to set up .spectral.yaml?

**To Reproduce**
openapi.yaml:
```
openapi: 3.0.0
info:
  title: test
  version: 1.0.0
  description: test
servers:
  - url: 'http://localhost:5000'
paths:
  '/test':
    get:
      operationId: test
      summary: A summary.
      description: |
        A description
      tags:
        - test
      responses:
        '200':
          description: OK

tags:
  - name: test
```

.spectral.yaml:
```
formats:
  - oas3
extends:
  -
    - https://raw.githubusercontent.com/openapi-contrib/style-guides/15d28d5/openapi.yml
    - 'off'
rules:
  operation-short-summary:
    description:
      Operation summary should be short and sweet, no full stops, and less than
      40 characters.
    severity: error
    recommended: true
    type: "style"
    given:
      $.paths.*[?(
      @property === 'get' ||
      @property === 'put' ||
      @property === 'post' ||
      @property === 'delete' ||
      @property === 'options' ||
      @property === 'head' ||
      @property === 'patch' ||
      @property === 'trace'
      )]
    then:
      - field: summary
        function: pattern
        functionOptions:
          notMatch: "\\."
      - field: summary
        function: length
        functionOptions:
          max: 40
```

Running `spectral lint openapi.yaml` with 5.7.1:
```
openapi.yaml
 12:16  error  operation-short-summary  Operation summary should be short and sweet, no full stops, and less than 40 characters.  paths./test.get.summary
```
With 6.3.0:
```
No results with a severity of 'error' or higher found!
```
If I remove the `off` part, 6.3.0 fails like 5.7.1.

**Expected behavior**
The rule to properly work and be evaluated as expected in 6.3.0 as it is in 5.7.1.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment (remove any that are not applicable):**
 - Library version: 5.7.1/6.3.0
 - OS: MacOS
 - Browser: N/A

--- a/packages/core/src/ruleset/__tests__/__fixtures__/severity/off-redeclared.ts
+++ b/packages/core/src/ruleset/__tests__/__fixtures__/severity/off-redeclared.ts
@@ -1,0 +1,15 @@
+import { RulesetDefinition } from '@stoplight/spectral-core';
+import shared from './shared';
+import { truthy } from '@stoplight/spectral-functions/src';
+
+export default {
+  extends: [[shared, 'off']],
+  rules: {
+    'overridable-rule': {
+      given: '$.foo',
+      then: {
+        function: truthy,
+      },
+    },
+  },
+} as RulesetDefinition;

--- a/packages/core/src/ruleset/__tests__/ruleset.test.ts
+++ b/packages/core/src/ruleset/__tests__/ruleset.test.ts
@@ -68,6 +68,18 @@ describe('Ruleset', () => {
       expect(getEnabledRules(rules)).toEqual(['overridable-rule']);
     });
 
+    it('given ruleset with extends set to off, should disable all rules but explicitly redeclared', async () => {
+      const { rules } = await loadRuleset(import('./__fixtures__/severity/off-redeclared'));
+      expect(Object.keys(rules)).toEqual([
+        'description-matches-stoplight',
+        'title-matches-stoplight',
+        'contact-name-matches-stoplight',
+        'overridable-rule',
+      ]);
+
+      expect(getEnabledRules(rules)).toEqual(['overridable-rule']);
+    });
+
     it('given nested extends with severity set to off', async () => {
       const { rules } = await loadRuleset(import('./__fixtures__/severity/off-proxy'));
       expect(Object.keys(rules)).toEqual([

--- a/packages/core/src/ruleset/mergers/rules.ts
+++ b/packages/core/src/ruleset/mergers/rules.ts
@@ -39,7 +39,10 @@ export function mergeRule(
       break;
     case 'object':
       if (existingRule !== void 0) {
-        Object.assign(existingRule, rule, { owner: existingRule.owner });
+        Object.assign(existingRule, rule, {
+          enabled: true,
+          owner: existingRule.owner,
+        });
       } else {
         assertValidRule(rule);
         return new Rule(name, rule, ruleset);

